src/lsp4clj/io_chan.clj

```diff
@@ -128,6 +128,10 @@
       (with-open [writer output] ;; close output when channel closes
         (loop []
           (when-let [msg (async/<!! messages)]
-            (write-message writer msg)
+            (try
+              (write-message writer msg)
+              (catch Throwable e
+                (async/close! messages)
+                (throw e)))
```

Comment:
I suppose this throw is enough to the developer know this was a JSON parse exception or something right?

Reply:
I think it depends a bit on the setup. Because this runs within a `core.async/thread`, the exception is probably given to the thread's uncaught exception handler. I think the default one will dump the stack trace to stderr, but many logging frameworks will allow to install a custom handler to log those. Definitely something people should do.

As a library, I figured we should stay un-opinionated here and just throw.

Reply:
Yeah, agreed